sexpr/lexer.go

```diff
@@ -0,0 +1,64 @@
+package sexpr
```

Comment:
The `sexpr` package is directly related to the `rule` package, what about moving it under `rule/sexpr` ?

Reply:
Good idea.